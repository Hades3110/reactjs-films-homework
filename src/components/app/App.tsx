import './app.scss'
import RoutePages from '../../router/router'

const App = () => (
        <RoutePages />
)

export default App
