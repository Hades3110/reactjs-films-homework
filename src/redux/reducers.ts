import { combineReducers } from "redux";
import filmListPageCountReducer from "./filmListCountAndCategories/reducer";
import filmListSortReducer from "./filmListSort/reducer";
import mainBannerReducer from "./mainBanner/reducer";

const rootReducer = combineReducers({
    pageCounter: filmListPageCountReducer,
    sortType: filmListSortReducer,
    mainBanner: mainBannerReducer
})

export default rootReducer
