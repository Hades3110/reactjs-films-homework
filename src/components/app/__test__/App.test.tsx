import { create } from "react-test-renderer";
import App from '../App';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from '../../../redux/reducers';

describe('App Component', () => {

  const store = createStore(rootReducer)

  it('App snapshot test', () => {
    const component = create(<Provider store={store}><App /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});