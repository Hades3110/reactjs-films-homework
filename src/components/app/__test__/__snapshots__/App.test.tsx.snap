// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App snapshot test 1`] = `
Array [
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <h1>
        FILMS
      </h1>
    </a>
    <form
      className="search"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        value=""
      />
      <button>
        <img
          alt="magnifying glass"
          src="test-file-stub"
        />
      </button>
    </form>
  </header>,
  "",
  <img
    className="spinner"
    src="test-file-stub"
  />,
  <main
    className="main"
  >
    <div
      className="filmFilter"
    >
      <div
        className="filmFilter__filter"
      >
        <button
          className="filmFilter__filter__btns activeCategories"
          onClick={[Function]}
        >
          Trending
        </button>
        <button
          className="filmFilter__filter__btns"
          onClick={[Function]}
        >
          Top Rated
        </button>
        <button
          className="filmFilter__filter__btns"
          onClick={[Function]}
        >
          Coming Soon
        </button>
      </div>
      <div
        className="filmFilter__arrange"
      >
        <button
          className="filmFilter__arrange__row activeSort"
          onClick={[Function]}
        >
          <div
            className="filmFilter__arrange__row__el"
          />
          <div
            className="filmFilter__arrange__row__el"
          />
          <div
            className="filmFilter__arrange__row__el"
          />
          <div
            className="filmFilter__arrange__row__el"
          />
        </button>
        <button
          className="filmFilter__arrange__oneColumn"
          onClick={[Function]}
        >
          <div
            className="filmFilter__arrange__oneColumn__el"
          />
          <div
            className="filmFilter__arrange__oneColumn__el"
          />
        </button>
      </div>
    </div>
    <img
      alt="spinner"
      className="spinner"
      src="test-file-stub"
    />
  </main>,
  <footer
    className="footer"
  >
    <div
      className="footer__name"
    >
      FILMS
    </div>
    <div
      className="footer__info"
    >
      Copyrigth Â© 2021 FILMS. HRANT GRISHYAN
    </div>
  </footer>,
]
`;
